h2o-py/h2o/automl/_estimator.py

```diff
@@ -404,11 +411,45 @@ def __validate_algo_parameters(self, algo_parameters):
     def __validate_frame(self, fr, name=None, required=False):
         return H2OFrame._validate(fr, name, required=required)
 
+    def __validate_distribution(self, distribution):
+        if is_type(distribution, str):
+            distribution = distribution.lower()
+            if distribution == "custom":
+                raise H2OValueError('Distribution "custom" has to be specified as a '
+                                    'dictionary with their respective parameters, e.g., '
+                                    '`dict(distribution = \"custom\", custom_distribution_func = \"...\"))`.')
+            return distribution
+        if is_type(distribution, dict):
+            dist = distribution["distribution"].lower()
```

Comment:
isn't it weird to use `distribution=dict(distribution=…)`? twice the same identifier? can confuse the user.
Usually you use something like`type` in that case:
```python
distribution=dict(type="tweedie", tweedie_power=0.5)
```

Reply:
Oops, this is from the original API spec, but now it seems obvious that this is a bit weird/redundant.  This is what we had:

 #### Python
```python
H2OAutoML(distribution="laplace")
H2OAutoML(distribution=dict(distribution="tweedie", tweedie_power=1.5))
H2OAutoML(distribution="tweedie")  #also works when you don't need to set additional args
```

#### R
```r
h2o.automl(distribution = "laplace", ...)
h2o.automl(distribution = list(distribution = "tweedie", tweedie_power = 1.5), ...)
h2o.automl(distribution = "tweedie", ...)  #also works when you don't need to set additional args
```

Reply:
Changed to use `list(type = `/`dict(type =` instead of `list(distribution = `/`dict(distribution=`.